Expensify/App
# Remove lingering border from AddressSearch component

### Details
Not my favorite solution, but it's a temporary workaround to compensate for the limitations of the library. 
Wrapped the AddressSearch component in a View to track it's hide and dynamically hide the lingering border.

cc @aldo-expensify @Luke9389 

### Fixed Issues
$ https://github.com/Expensify/App/issues/5810

### Tests
1. On NewDot, navigate to `/bank-account/company`
2. Type in an address that does not exist, e.g. `asdasdas`
3. Verify that there is no lingering border below the input
4. Now, type in an address that exists and select it from the dropdown.
5. Click on the input again and verify that there is no lingering border.

### QA Steps
Steps above.

### Tested On


### Screenshots
#### Web

https://user-images.githubusercontent.com/22219519/139493256-ed0366da-1137-48df-923e-02db7f618d5a.mov

#### Mobile Web
The component is failing to connect to www.expensify.com.dev on the simulator and returns no address results. We might be missing some config here. This is happening on the main branch as well.

#### Desktop

https://user-images.githubusercontent.com/22219519/139493277-38d3f14d-26c5-447e-9dba-3fd1bef7c93c.mov

#### iOS

https://user-images.githubusercontent.com/22219519/139493296-ea2fa9af-a629-4a2d-afd1-bbd1a1192895.mov

#### Android

https://user-images.githubusercontent.com/22219519/139493305-eadde9eb-d548-4ea8-9b80-2a6177470478.mov
